build.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+// Run "cargo build -vv" if you want to see gn output.
+// TODO For the time being you must set an env var DENO_BUILD_PATH
+// which might be `pwd`/out/debug or `pwd`/out/release.
+// TODO Currently DENO_BUILD_PATH must be absolute.
+// TODO Combine DENO_BUILD_PATH and OUT_DIR.
+
+use std::env;
+use std::process::Command;
+
+fn main() {
+  let mode = env::var("PROFILE").unwrap();
+  let deno_build_path = env::var("DENO_BUILD_PATH").unwrap();
+
+  let status = Command::new("./tools/setup.py")
+    .env("DENO_BUILD_PATH", &deno_build_path)
+    .env("DENO_BUILD_MODE", &mode)
+    .status()
+    .expect("setup.py failed");
+  assert!(status.success());
+
+  // These configurations must be outputted after tools/setup.py is run.
+  println!("cargo:rustc-env=OUT_DIR={}", deno_build_path);
```

Comment:
I don't think it's a good idea to override this cargo built-in enviroment variable, if it works at all.
The purpose of it is not entirely clear to me:
* if you want to make cargo use a different output directory -- it isn't entirely working.
* if you want to teach msg.rs where msg_generated.rs lives, use an env var that doesn't conflict with a cargo built-in.

Reply:
Good point. I've changed it to be DENO_BUILD_PATH and added TODOs to eventually combine with OUT_DIR